pkg/ipamd/ipamd.go

```diff
@@ -1033,20 +1037,25 @@ func (c *IPAMContext) shouldRemoveExtraPrefixes() int {
 
 	total, used, _ := c.dataStore.GetStats()
 	available := total - used
-	var shouldRemoveExtra bool
 
-	warmTarget := (c.warmPrefixTarget + 1)
+	freePrefixes := c.dataStore.GetFreePrefixes()
+        _, numIPsPerPrefix, _ := datastore.GetPrefixDelegationDefaults()
 
-	shouldRemoveExtra = available >= (warmTarget)*c.maxPrefixesPerENI
-	if shouldRemoveExtra {
-		freePrefixes := c.dataStore.GetFreePrefixes()
+	over = max(freePrefixes-c.warmPrefixTarget, 0)
 
-		over = max(freePrefixes-c.warmPrefixTarget, 0)
-		logPoolStats(total, used, c.maxIPsPerENI, c.enableIpv4PrefixDelegation)
-		log.Debugf("It might be possible to remove extra prefixes because available (%d) >= (Prefix target + 1 (%d) + 1) * prefixesPerENI (%d)", available, warmTarget, c.maxPrefixesPerENI)
+	//if warm prefix target is 0, we should not make available to reach 0 since reconciler will enter into a churn
+	// of allocating and freeing prefixes similar to issue. This will add continous EC2 calls on nodes which have
+	// no prefixes used. Hence better to handle it. 
+	// [ENI allocates/frees in loop with custom networking and WARM_ENI_TARGET=0 #1451]
+	// But Will sync up internally with the team and see if need to even support WARM_ENI_TARGET = 0 and WARM_PREFIX_TARGET = 0
+	if over > 0 && ((available - (over * numIPsPerPrefix)) <= 0) && c.warmPrefixTarget == 0 {
```

Comment:
I have discussed this internally and having WARM_{PREFIX,ENI}_TARGET = 0 and WARM_IP_TARGET = 0 will not help adding IPs to datastore. So in the follow up PR, this section will be removed and also `(c.warmENITarget == 0 && available == 0)` in `isDataStoreLow()`. Also will be adding documentation on the recommended configuration setting. WARM_IP_TARGET will override WARM_{PREFIX,ENI}_TARGET as expected but if there is WARM_{PREFIX,ENI}_TARGET then it has to be non-zero. This will fix #1451 which will cause unwanted churn with ENI creates and deletes.

Reply:
Since, we're introducing `WARM_PREFIX_TARGET` for the very first time we can avoid this scenario by not allowing `0` for this env variable. Our default is going to be `1` anyways and we will be able to do away with this additional check. As discussed offline, we can take it up in a follow up PR.

Reply:
Yes Apurup, For warm_prefix_target, lets default it to 1. But for warm_eni_target there might be some Cx using it so lets discuss more and see how we can decommission the support since it has few issues where IPAMD silently keeps adding and removing the ENI.